kclvm/parser/src/session/mod.rs

```diff
@@ -57,9 +57,110 @@ impl ParseSession {
         self.add_parse_err(ParseError::Message {
             message: msg.to_string(),
             span,
+            fix_info: None,
         });
     }
 
+    #[inline]
+    pub fn struct_span_error_with_suggestions(
+        &self,
+        msg: &str,
+        span: Span,
+        suggestion_text: Option<String>,
+        replacement_text: Option<String>,
+    ) {
+        let code_line = self.get_line_content_from_span(span);
+        let modified_code_line =
+            self.generate_modified_code_line(msg, &code_line, replacement_text.as_deref());
+
+        let fix_info = Some(FixInfo {
+            suggestion: suggestion_text,
+            replacement: Some(modified_code_line),
```

Comment:
Why replace the whole line instead of the wrong position

Reply:
you can use the `replacement_text` as `replacement`.
and in quick_fix, just replace the diags postion with replacement.
do not need to get the whole line, replace the `replacement_text`, generate new whole line, and replce whole line in quickfix

Reply:
Changes implemented in the latest commit, instead of modifying the whole code line, a modified range is passed to quick fix. @He1pa Please review.

cc @Peefy